assets/js/blocks/cart-checkout/payment-methods/express-payment-methods.js

```diff
@@ -48,32 +55,71 @@ const ExpressPaymentMethods = () => {
 			setPaymentStatus,
 		]
 	);
+
+	/**
+	 * onExpressPaymentClose should be triggered when the express payment process is cancelled or closed.
+	 *
+	 * This restores the active method and returns the state to pristine.
+	 */
 	const onExpressPaymentClose = useCallback( () => {
+		setPaymentStatus().pristine();
 		setActivePaymentMethod( previousActivePaymentMethod.current );
 		if ( previousPaymentMethodData.current.isSavedToken ) {
 			setPaymentStatus().started( previousPaymentMethodData.current );
 		}
 	}, [ setActivePaymentMethod, setPaymentStatus ] );
-	const paymentMethodIds = Object.keys( paymentMethods );
+
+	/**
+	 * onExpressPaymentError should be triggered when the express payment process errors.
+	 *
+	 * This shows an error message then restores the active method and returns the state to pristine.
+	 */
+	const onExpressPaymentError = useCallback(
```

Comment:
Error handling is new and allows us to close express payment, restore everything, and show the notice. This can be tested by using Stripe 3d secure and clicking "Fail".